### PR TITLE
Flagon

### PR DESCRIPTION
**Pull request by jonathan-fantham**

### CODE REVIEW

**Comment by jazminschroeder** [comment]
:clap:  I like it!

File: _posts/2015-03-31-flagon.markdown [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+{% endhighlight %}
+
+Flagon will call those two methods to work out whether a flag is on or off and your application can be blissfully ignorant about how or where the flags are defined in development, test or production.
+
+For example, you could create an active record model with two attributes "name" and "value" and implement the two interface methods like so:
+
+{% highlight ruby %}
+class Flag < ::ActiveRecord::Base
+  def self.exists?(flag_name)
+    where(name: flag_name).count > 0
```
Comment (cbandy) [inline _posts/2015-03-31-flagon.markdown]: `count` is a good way to do this; [`exists?`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F) is better. Not sure what to do about the collision.

Comment (cbandy) [inline _posts/2015-03-31-flagon.markdown]: `count` is a not good way to do this; [`exists?`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F) is better. Not sure what to do about the collision.

Comment (jonathan-fantham) [inline _posts/2015-03-31-flagon.markdown]: is there a better way to do it that isn't exists? E.g. `where(name: flag_name).first.present?`

Comment (cbandy) [inline _posts/2015-03-31-flagon.markdown]: Probably the interface should be `flag_exists?` and `get_flag`.

Comment (jonathan-fantham) [inline _posts/2015-03-31-flagon.markdown]: maybe `super(name: flag_name)`

Comment (cbandy) [inline _posts/2015-03-31-flagon.markdown]: I mean, if you want an AR model to be trivial.

Comment (jonathan-fantham) [inline _posts/2015-03-31-flagon.markdown]: Is it worth making another release of flagon to suit this example?

Comment (cbandy) [inline _posts/2015-03-31-flagon.markdown]: > maybe `super(name: flag_name)`

That might work. Of course, `Flag` isn't acting like AR any more. :wink:

Comment (cbandy) [inline _posts/2015-03-31-flagon.markdown]: > Is it worth making another release of flagon to suit this example?

That's up to you!

Does this work?

``` ruby
class Flag < ::ActiveRecord::Base
  module Loader
    def self.exists?(flag_name)
      Flag.exists?(name: flag_name)
    end

    def self.get_flag(flag_name)
      flag = Flag.where(name: flag_name).first
      ...
    end
  end
end

Flagon.init Flag::Loader
```

Comment (jonathan-fantham) [inline _posts/2015-03-31-flagon.markdown]: I wouldn't think so, but I can try it later when I get a chance

**Pull request merged**